DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -1551,6 +1616,43 @@ private void GameBroadcastChannel_CTCPReceived(object sender, ChannelCTCPEventAr
                 int skillLevel = int.Parse(splitMessage[11]);
                 string mapHash = splitMessage[12];
 
+                bool[] checkboxValues = null;
+                int[] dropdownIndices = null;
+
+                // Games with different versions may have different option counts, so ignore
+                if (gameVersion == ProgramConstants.GAME_VERSION)
+                {
+                    // packed checkbox values
+                    if (!string.IsNullOrEmpty(splitMessage[13]))
+                    {
+                        string[] checkboxStrings = splitMessage[13].Split(',');
+                        int[] packedCheckboxes = new int[checkboxStrings.Length];
+                        for (int i = 0; i < checkboxStrings.Length; i++)
+                            packedCheckboxes[i] = int.Parse(checkboxStrings[i]);
+
+                        int localCheckboxCount = gameLobby.CheckBoxes.Count(cb => cb.BroadcastToLobby);
+                        checkboxValues = new bool[localCheckboxCount];
+                        for (int i = 0; i < localCheckboxCount; i++)
+                        {
+                            int packedIndex = i / 32;
+                            int bitIndex = i % 32;
+                            if (packedIndex < packedCheckboxes.Length)
+                                checkboxValues[i] = (packedCheckboxes[packedIndex] & (1 << bitIndex)) != 0;
+                        }
```

Comment:
could you explain what is happening here? I see bitpacking which is a clever optimisation, but apart from that I have no idea